### ISSUE TITLE
0.11.0 release checklist

### ISSUE THREAD

**Issue by patrickjahns** [owncloud/user_ldap#174]
### Reasons

Full list of PRs is here https://github.com/owncloud/user_ldap/milestone/11?closed=1 but summarised below.

 - Primarily, we moved sync login into core and out of LDAP https://github.com/owncloud/user_ldap/pull/156  @jvillafanez :x: https://github.com/owncloud/user_ldap/issues/174#issuecomment-382061868
 - Killed all the old ajax.php and moved to controllers https://github.com/owncloud/user_ldap/pull/199
 - Much improved acceptance and UI automated tests
 - UUIDs are not trimmed https://github.com/owncloud/user_ldap/pull/142  @jvillafanez ❗️ https://github.com/owncloud/user_ldap/issues/174#issuecomment-382308470
 - JSON output for list config command https://github.com/owncloud/user_ldap/pull/185  @jvillafanez ✅ (https://github.com/owncloud/user_ldap/issues/174#issuecomment-382314626)
 - Make sure new encryption keys use password with LDAP correctly https://github.com/owncloud/user_ldap/issues/177  @jvillafanez ✅ 

### QA
- [x] Create release branch - @patrickjahns - https://github.com/owncloud/user_ldap/tree/release/0.11.0 
- [x] Create "$version Testplan" @voroyam https://github.com/owncloud/user_ldap/issues/230
- [ ] Test and document in testplan => Final "Go" from QA (set 'QA-ready' label)

### Marketing
- [ ] ~~Check/adapt info.xml stuff~~
- [ ] ~~Check/adapt screenshot~~
- [ ] ~~Blog post or other communication if applicable~~

### Documentation
- [ ] Changes in the docs needed?

### Build & Marketplace release

- [x] Bump version in info.xml
- [x] Update CHANGELOG.md - Change CHANGELOG.md 'Unreleased' to '$version' @patrickjahns
- [x] Create github tag for '$version' @patrickjahns
- [x] Build and sign tarball from github tag '$version' @patrickjahns
- [x] Upload to marketplace @patrickjahns
- [x] Send mail to release-coordination with link to this ticket and to the changelog @patrickjahns
- [x] Rename QA milestone to '$version' @patrickjahns
- [ ] Add a new 'Unreleased' section in CHANGELOG.md @patrickjahns
- [x] Create PR to merge changes back to `master` @patrickjahns

**Comment by patrickjahns** [comment]
Pre-release tarball available at https://cloud.owncloud.com/index.php/s/Iuu1tBK997GxqLE

NOTE: this tarball is **not** signed as it is pre-release

**Comment by jvillafanez** [comment]
* [ ] BUG single user syncing considers the user as missing:
    1. Setup ownCloud with user_ldap
    2. Run `occ user:sync "OCA\User_LDAP\User_Proxy"` to sync all the users
    3. Run `occ user:sync -u <uid> "OCA\User_LDAP\User_Proxy"`
    ```
    root@e735b7973972:/opt/owncloud# sudo -u www-data ./occ user:sync -u 9c7b4656-d008-1036-8555-dd0c704ad877 "OCA\User_LDAP\User_Proxy"If unknown users are found, what do you want to do with their accounts? (removing the account will also remove its data)
      [0] disable
      [1] remove
      [2] ask later
     > 2
    Syncing 9c7b4656-d008-1036-8555-dd0c704ad877 ...
    These accounts that are no longer available in the backend:
    9c7b4656-d008-1036-8555-dd0c704ad877
    What do you want to do with their accounts? (removing the account will also remove its data)
      [0] disable
      [1] remove
     > 0
    Disabling accounts
    9c7b4656-d008-1036-8555-dd0c704ad877, aaliyah aañote, aannote@pmail.com disabled
    ```
* [ ] BUG `--showCount` doesn't seem to work properly (no difference with the option set regarding normal syncing)
    ```
    root@e735b7973972:/opt/owncloud# sudo -u www-data ./occ user:sync -c "OCA\User_LDAP\User_Proxy"
    If unknown users are found, what do you want to do with their accounts? (removing the account will also remove its data)
      [0] disable
      [1] remove
      [2] ask later
     > 2
    Analysing known accounts ...
     1002 [============================]

    No removed users have been detected.


    Inserting new and updating all known users from OCA\User_LDAP\User_Proxy ...
     1001/1001 [============================] 100%
    ```

Working:
* `--seen-only`
* `--re-enable`
* `--missing-account-action`
* `--seen-only` + `--re-enable`
* user sync during login

**Comment by jvillafanez** [comment]
https://github.com/owncloud/user_ldap/pull/142 is untesteable. According to http://ldapwiki.com/wiki/EntryUUID the entryUUID value is automatically generated by the server and can't be modified by the user.
I've also tried to use the uid but I'm having a lot of problems trying to include a whitespace or newline at the beginning or end of the uid. I'm not sure if it's due to the uid being part of the dn. Other attributes will likely cause trouble.

**Comment by jvillafanez** [comment]
https://github.com/owncloud/user_ldap/pull/185 looks fine, although with multiple ldap configuration it returns multiple jsons. I'd rather show only one json with multiple entries such as
```
{
  "": {"ldapHost":"....", "ldapPort":"2222", ....},
  "s01": {"ldapHost":...", "ldapPort":"4444", ....}
}
```
Same format for just one configuration.

**Comment by butonic** [comment]
@jvillafanez you were testing with AD? and objectguid was in the additional search attributes?

**Comment by jvillafanez** [comment]
It was openLDAP with the default configuration, just added the attributes for the quota and mail (group membership also changed to "member"). No additional search attributes were set.

**Comment by butonic** [comment]
Testing against samba4 as AD DC:
```
➜  core git:(stable10) ✗ ./occ ldap:show-config                                                                            
+-------------------------------+-------------------------------------------------------------------------------------------+
| Configuration                 | s01                                                                                       |
+-------------------------------+-------------------------------------------------------------------------------------------+
| hasMemberOfFilterSupport      | 0                                                                                         |
| hasPagedResultSupport         |                                                                                           |
| homeFolderNamingRule          |                                                                                           |
| lastJpegPhotoLookup           | 0                                                                                         |
| ldapAgentName                 | cn=Administrator,cn=users,dc=home,dc=oberdreyer,dc=de                                     |
| ldapAgentPassword             | ***                                                                                       |
| ldapAttributesForGroupSearch  |                                                                                           |
| ldapAttributesForUserSearch   | objectguid                                                                                |
| ldapBackupHost                |                                                                                           |
| ldapBackupPort                |                                                                                           |
| ldapBase                      | dc=home,dc=oberdreyer,dc=de                                                               |
| ldapBaseGroups                | cn=groups,dc=home,dc=oberdreyer,dc=de                                                     |
| ldapBaseUsers                 | cn=users,dc=home,dc=oberdreyer,dc=de                                                      |
| ldapCacheTTL                  | 600                                                                                       |
| ldapConfigurationActive       | 1                                                                                         |
| ldapDynamicGroupMemberURL     |                                                                                           |
| ldapEmailAttribute            | mail                                                                                      |
| ldapExperiencedAdmin          | 0                                                                                         |
| ldapExpertUUIDGroupAttr       |                                                                                           |
| ldapExpertUUIDUserAttr        | objectguid                                                                                |
| ldapExpertUsernameAttr        |                                                                                           |
| ldapGroupDisplayName          | cn                                                                                        |
| ldapGroupFilter               | (&(|(objectclass=group)))                                                                 |
| ldapGroupFilterGroups         |                                                                                           |
| ldapGroupFilterMode           | 1                                                                                         |
| ldapGroupFilterObjectclass    | group                                                                                     |
| ldapGroupMemberAssocAttr      | member                                                                                    |
| ldapHost                      | 127.0.0.1                                                                                 |
| ldapIgnoreNamingRules         |                                                                                           |
| ldapLoginFilter               | (&(|(objectclass=person))(|(samaccountname=%uid)(|(mailPrimaryAddress=%uid)(mail=%uid)))) |
| ldapLoginFilterAttributes     |                                                                                           |
| ldapLoginFilterEmail          | 1                                                                                         |
| ldapLoginFilterMode           | 0                                                                                         |
| ldapLoginFilterUsername       | 1                                                                                         |
| ldapNestedGroups              | 0                                                                                         |
| ldapOverrideMainServer        |                                                                                           |
| ldapPagingSize                | 500                                                                                       |
| ldapPort                      | 389                                                                                       |
| ldapQuotaAttribute            |                                                                                           |
| ldapQuotaDefault              |                                                                                           |
| ldapTLS                       | 0                                                                                         |
| ldapUserDisplayName           | cn                                                                                        |
| ldapUserDisplayName2          |                                                                                           |
| ldapUserFilter                | (|(objectclass=person))                                                                   |
| ldapUserFilterGroups          |                                                                                           |
| ldapUserFilterMode            | 1                                                                                         |
| ldapUserFilterObjectclass     | person                                                                                    |
| ldapUuidGroupAttribute        | auto                                                                                      |
| ldapUuidUserAttribute         | auto                                                                                      |
| turnOffCertCheck              | 0                                                                                         |
| useMemberOfToDetectMembership | 1                                                                                         |
+-------------------------------+-------------------------------------------------------------------------------------------+
➜  core git:(stable10) ✗ ./occ user:sync "OCA\User_LDAP\User_Proxy" -m disable -vvv -u 1415264A-5842-45F6-897F-E8B26CF68104
Syncing 1415264A-5842-45F6-897F-E8B26CF68104 ...

➜  core git:(stable10) ✗ ./occ:user delete 1415264A-5842-45F6-897F-E8B26CF68104
zsh: datei oder Verzeichnis nicht gefunden: ./occ:user
➜  core git:(stable10) ✗ ./occ user:delete 1415264A-5842-45F6-897F-E8B26CF68104
User with uid '1415264A-5842-45F6-897F-E8B26CF68104', display name 'jfd', email '' was deleted
➜  core git:(stable10) ✗ ./occ user:sync "OCA\User_LDAP\User_Proxy" -m disable -vvv -u 1415264A-5842-45F6-897F-E8B26CF68104
Syncing 1415264A-5842-45F6-897F-E8B26CF68104 ...

➜  core git:(stable10) ✗ ./occ user:sync "OCA\User_LDAP\User_Proxy" -m disable -vvv                                        
Analysing known accounts ...
    7 [============================] < 1 sec 20.0 MiB

No removed users have been detected.


Inserting new and updating all known users from OCA\User_LDAP\User_Proxy ...
    5 [============================] < 1 sec 20.0 MiB

➜  core git:(stable10) ✗ ./occ user:sync "OCA\User_LDAP\User_Proxy" -m disable -vvv -u 1415264A-5842-45F6-897F-E8B26CF68104
Syncing 1415264A-5842-45F6-897F-E8B26CF68104 ...
```

@jvillafanez for openldap add entryuuid to the search attributes.

**Comment by jvillafanez** [comment]
My point is that it's unlikely that people will include that attribute unless told, and there is no warning message that tell the user that he need to include the attribute. This implies that normal users will open tickets feeling that the behaviour is buggy because the command doesn't do what they expect.

I'm not sure if it's easy to include the attribute automatically or transparently without user intervention, but it's something we have to try.
